Source/ASRunLoopQueue.h

```diff
@@ -48,6 +55,30 @@ AS_SUBCLASSING_RESTRICTED
 
 @end
 
+AS_SUBCLASSING_RESTRICTED
+@interface ASCATransactionQueue : ASAbstractRunLoopQueue
+
+@property (nonatomic, readonly) BOOL isEmpty;
+@property (nonatomic, readonly) BOOL disabled;
+/**
+ * The queue to run on main run loop before CATransaction commit.
+ *
+ * @discussion this queue will run after ASRunLoopQueue and before CATransaction commit
+ * to get last chance of updating/coalesce info like interface state.
+ * Each node will only be called once per transaction commit to reflect interface change.
+ */
++ (ASCATransactionQueue *)sharedQueue;
+
+- (void)enqueue:(id<ASCATransactionQueueObserving>)object;
+
+/**
+ * @abstract Apply a node's interfaceState immediately rather than adding to the queue.
+ */
+- (void)disableInterfaceStateCoalesce;
```

Comment:
I would just call this method `-disable`, since this queue isn't supposed to know anything about interface states per-se.